### PR TITLE
foxglove-websocket: Add version 1.0.0

### PR DESCRIPTION
**Pull request by achim-k**
Specify library name and version:  **foxglove-websocket/1.0.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by achim-k** [comment]
@jcar87 is there anything I can do to help getting this PR in?

File: recipes/foxglove-websocket/all/conanfile.py [hunk 1]
```diff
@@ -11,30 +17,98 @@ class FoxgloveWebSocketConan(ConanFile):
...
+            self.options.rm_safe("shared")
+            self.options.rm_safe("fPIC")
+        elif self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def build(self):
+        if self.version >= Version("1.0.0"):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
```
Comment (franramirez688) [inline recipes/foxglove-websocket/all/conanfile.py:97]: Should we remove the `0.0.1` version?

Comment (achim-k) [reply recipes/foxglove-websocket/all/conanfile.py:97]: Not sure if I understand correctly, do you mean to remove the `0.0.1` version from this recipe completely?

File: recipes/foxglove-websocket/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-      server.stop();
-    });
-    server.run();
+    const auto logHandler = [](foxglove::WebSocketLogLevel, char const* msg) {
+      std::cout << msg << std::endl;
+    };
+    foxglove::ServerOptions serverOptions;
+    foxglove::Server<foxglove::WebSocketNoTls> server("example", logHandler, serverOptions);
+    server.start("127.0.0.1", 0);
+    server.stop();
```
[the remark this answers is not in the document]
Comment (achim-k) [reply recipes/foxglove-websocket/all/test_package/test_package.cpp:14]: I believe it's fine to just instantiate the server and not actually run it. Testing that it compiles is sufficient, I will change that.

File: recipes/foxglove-websocket/all/conanfile.py (hunk 1 above)
Comment (franramirez688) [reply recipes/foxglove-websocket/all/conanfile.py:97]: Yeah, I mean, we're skipping the `0.0.1` version from the `layout`, `generate`, `build`, etc. steps, so why keep it there? @uilianries what do you think? In my opinion, let's avoid that `if` conditions if we're not going to build `0.0.1` anymore.

File: recipes/foxglove-websocket/all/test_package/test_package.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/foxglove-websocket/all/test_package/test_package.cpp]: It seems be a good deal removing 0.0.1 and running a grep here, I didn't find other recipes using it. Removing that older version would simplify it a lot! Please, remove 0.0.1 :)

[the remark this answers is not in the document]
Comment (franramirez688) [reply recipes/foxglove-websocket/all/test_package/test_package.cpp]: Great! So let's remove it. We'll get a more simplified recipe 👏

[the remark this answers is not in the document]
Comment (franramirez688) [reply recipes/foxglove-websocket/all/test_package/test_package.cpp]: Well, the correct thread was https://github.com/conan-io/conan-center-index/pull/16500#discussion_r1189958274 😆

File: recipes/foxglove-websocket/all/conanfile.py (hunk 1 above)
Comment (franramirez688) [reply recipes/foxglove-websocket/all/conanfile.py:97]: https://github.com/conan-io/conan-center-index/pull/16500#discussion_r1190002679

**Review by franramirez688 — APPROVED** [review]
Looks good to me! Great job!

**Review by SSE4 — APPROVED** [review]

**Pull request merged**